manifests/kustomize/third-party/metacontroller/base/stateful-set.yaml

```diff
@@ -18,26 +18,28 @@ spec:
         sidecar.istio.io/inject: "false"
     spec:
       containers:
-      - command:
-        - /usr/bin/metacontroller
-        - --logtostderr
-        - -v=4
-        - --discovery-interval=20s
-        image: metacontroller/metacontroller:v0.3.0
-        imagePullPolicy: Always
-        name: metacontroller
-        ports:
-        - containerPort: 2345
-        resources:
-          limits:
-            cpu: "4"
-            memory: 4Gi
-          requests:
-            cpu: 500m
-            memory: 1Gi
-        securityContext:
-          allowPrivilegeEscalation: true
-          privileged: true
+        - resources:
+            limits:
+              cpu: '1'
+              memory: 1Gi
+            requests:
+              cpu: 100m
+              memory: 256Mi
+          command:
+            - /usr/bin/metacontroller
+            - --zap-log-level=4
+            - '--discovery-interval=3600s' # less insane than 10 seconds
```

Comment:
What does this arg do? If a new namespace is created can required resources get created right away?

Reply:
Yes the resources are created right away in my installation.

From the documentation  https://metacontroller.github.io/metacontroller/guide/configuration.html?highlight=discovery-interval#command-line-flags

```
--discovery-interval | How often to refresh discovery cache to pick up newly-installed resources (e.g. --discovery-interval=10s). I think that means CRDs. A low value creates too much log spam and on start the cache is refreshed anyway. I am also fine with 10 minutes if that is what you want.

--zap-log-level | ZapGws  Level to configure the verbosity of logging. Can be one of ‘debug’,  ‘info’, ‘error’, or any integer value > 0 which corresponds to custom  debug levels of increasing verbosity(e.g. --zap-log-level=5).  Level 4 logs Metacontroller's interaction with the API server. Levels 5  and up additionally log details of Metacontroller's invocation of  lambda hooks. See the troubleshooting guide for more.
```

What you are worrying about is https://metacontroller.github.io/metacontroller/api/compositecontroller.html?highlight=discovery%20cache#resync-period but even that can be set to a high value in the compositecontroller "Sometimes you may want to sync periodically even if nothing has changed in the Kubernetes API objects"

Reply:
> /approve
> 
> Thank you @juliusvonkohout

Thank you too.

"To complete the pull request process, please ask for approval from bobgy after the PR has been reviewed." So i think @zijianjoy approval is not enough @bobgy

Reply:
Is this detailed enough, or should i test something for you ?